### ISSUE TITLE
Support Alpine linux

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/libuv-package#23]
Requires compiling for musl, not glibc.

**Workaround**
Users can remove libuv from their app and rely on the system's installed version of libuv instead.
```xml
<ItemGroup>
  <PackageReference Include="Libuv" Version="1.10.0" ExcludeAssets="Native" />
</ItemGroup>
```
Then, install libuv on host machines.
```
apk add libuv
ln -s /usr/lib/libuv.so.1 /usr/lib/libuv.so
```

**Comment by muratg** [comment]
cc @glennc

**Comment by Petermarcu** [comment]
Can you flip back to using sockets to unblock temporarily?

**Comment by natemcmaster** [comment]
At the moment, Kestrel's default is still sockets in 2.1 preview1 bits. This default is subject to change before RTM. https://github.com/aspnet/KestrelHttpServer/issues/2220

**Comment by Petermarcu** [comment]
Is see, I haven't seen the error for Alpine but what is blocking then? Is it that tests are failing on Alpine because they are explicitly running on libuv?

**Comment by natemcmaster** [comment]
Right. When I say it's a blocker, I mean blocking moving forward with Alpine in a supported way. We can disable tests, and produce a shared framework is broken for several key scenarios on Alpine...but I wouldn't want to ship that.

**Comment by Petermarcu** [comment]
We need to unblock product construction first. Alpine is already noted as a best effort OS and not in our supported list. Could you just disable the tests on Alpine? I also thought tests were not critical path for product construction

**Comment by natemcmaster** [comment]
Yes we can do those things. I updated my PR to the CLI last night to skip tests and pull our portable sharedfx even though it won't work. https://github.com/dotnet/cli/pull/8341

**Comment by Petermarcu** [comment]
Sounds good. Thanks! Hopefully Alpine will be off the blocking list then.

**Comment by natemcmaster** [comment]
Punting to backlog as we want to make sockets the default transport in Kestrel. Users could workaround for now by installing the libuv apk.

**Comment by Petermarcu** [comment]
Did we make the call on switching to Sockets as the default for Kestrel? Can we drop the libuv package from the nuget dependency graph too?

**Comment by natemcmaster** [comment]
Yes. I’ll share our design meeting notes on Monday

Sent from Outlook<https://aka.ms/qtex0l> for iOS

**Comment by natemcmaster** [comment]
[Design meeting notes from March 2, 2018](https://gist.github.com/natemcmaster/43a3980b724fb0b167844fb574df75dc)

See also https://github.com/aspnet/KestrelHttpServer/issues/2360, https://github.com/aspnet/KestrelHttpServer/issues/2361

**Comment by muratg** [comment]
@DamianEdwards Alpine Linux

**Comment by Petermarcu** [comment]
Do you want to support the All package and libuv transport on Alpine? If so, wouldn't you still need this issue?

**Comment by natemcmaster** [comment]
@muratg we should still probably consider this. Although you can workaround it with a globally installed libuv, it's not always easy to get that setup correctly.

**Comment by ghost** [comment]
@Petermarcu, @natemcmaster, I have built libuv binary on Alpine Linux with https://github.com/aspnet/libuv-build/pull/34. Can this be shipped with 2.1? We can also build libuv for arm32 to match the dotnet-core supported platform matrix: https://github.com/aspnet/Universe/issues/554.

**Comment by natemcmaster** [comment]
We are currently not planning on re-building the Libuv package to support Alpine in 2.1. However, we are still planning to enable ASP.NET Core for Alpine (already available in nightly builds, see <https://hub.docker.com/r/microsoft/dotnet-nightly>.) In 2.1, Kestrel will no longer rely on Libuv by default, so this issue isn't essential to getting ASP.NET Core + Alpine working.

We don't have plans to support arm32 in ASP.NET Core, at the moment.

**Comment by muratg** [comment]
Moved this to 3.0.0
